spec/expression.dd

```diff
@@ -2153,6 +2155,29 @@ $(GNAME ArrayLiteral):
         }
         ---
 
+    $(P An array literal is not GC allocated if:)
+
+    * It initializes or assigns to a static array.
+    * It initializes a $(DDSUBLINK spec/attribute, scope, `scope`) slice.
+
```

Comment:
If it's used in a `foreach` where the element type isn't `ref`, it's also not GC allocated

```D
foreach (i; [6, 7]) {} // converted to static array
foreach (ref i; [8, 9]) {} // GC allocated because `ref` may escape
```

Reply:
Thanks, I also added 2 other cases.